code/game/gamemodes/wizard/artefact.dm

```diff
@@ -245,9 +246,30 @@
 		current_owner.update_icons()
 
 /obj/item/scrying/attack_self(mob/user as mob)
-	to_chat(user, "<span class='notice'> You can see...everything!</span>")
-	visible_message("<span class='danger'>[user] stares into [src], [user.p_their()] eyes glazing over.</span>")
-	user.ghostize(1)
+	if(!in_use)
+		in_use = TRUE
+		ADD_TRAIT(user, SCRYING, SCRYING_ORB)
+		user.add_atom_colour(COLOR_BLUE, ADMIN_COLOUR_PRIORITY) // stolen spirit rune code
+		user.visible_message("<span class='notice'>[user] stares into [src], [user.p_their()] eyes glazing over.</span>",
+						"<span class='danger'> You stare into [src], you can see the entire universe!</span>")
+		ghost = user.ghostize(TRUE)
+		ghost.name = "Magic Spirit of [ghost.name]"
+		ghost.color = COLOR_BLUE
+		while(!QDELETED(user))
```

Comment:
Prob want to check after this loop if they were qdeleted or not. If they are then a lot of the logic below doesn't make sense.

Reply:
I don't quite feel comfortable yet with fully component-izing stuff yet. Also I added that check but I'm not sure if thats what you want? As far as I understand (with this being recycled spirit realm code), the while(!QDELETED(user)) is just meant to keep looping so that the checks in the loop can eventually break, so I'm not quite sure how helpful the if(QDELETED(user)) check would be, but it's been added

Reply:
No worries!
The if check should suffice yeah. Removing a trait or setting a color on an object that is already deleted doesn't make much sense.